pkg/networkdevice/profile/profiledefinition/metrics.go

```diff
@@ -47,8 +47,9 @@ type SymbolConfig struct {
 	ExtractValue         string         `yaml:"extract_value,omitempty" json:"extract_value,omitempty"`
 	ExtractValueCompiled *regexp.Regexp `yaml:"-" json:"-"`
 
-	MatchPattern         string         `yaml:"match_pattern,omitempty" json:"match_pattern,omitempty"`
-	MatchValue           string         `yaml:"match_value,omitempty" json:"match_value,omitempty"`
+	// MatchPattern/MatchValue are not exposed as json (UI) since ExtractValue can be used instead
+	MatchPattern         string         `yaml:"match_pattern,omitempty" json:"-"`
+	MatchValue           string         `yaml:"match_value,omitempty" json:"-"`
```

Comment:
We don't every parse these strictly, do we? Nothing should complain if you get a json struct with "match_pattern" or "match_value" set?

Reply:
It might also be worth adding a comment somewhere to make it clear that JSON format is for the UI and YAML is for internal use, just to explain why we skip some things in json but not in yaml.

Reply:
> We don't every parse these strictly, do we? Nothing should complain if you get a json struct with "match_pattern" or "match_value" set?

The parser json parser shouldn't complain since those fields are ignored (`json:"-"`)
but the jsonschema won't allow those fields since the schema doesn't allow additional properties and will raise an error (e.g. before storage in RC).

Reply:
> It might also be worth adding a comment somewhere to make it clear that JSON format is for the UI and YAML is for internal use, just to explain why we skip some things in json but not in yaml.


Very good point, added here https://github.com/DataDog/datadog-agent/pull/19751/commits/931f0b5c90c5b74f6c3897fd9e37c83f1852e128